packages/iov-ethereum/src/ethereumconnection.ts

```diff
@@ -294,4 +382,12 @@ export class EthereumConnection implements BcpConnection {
   public liveTx(_: BcpTxQuery): Stream<ConfirmedTransaction> {
     throw new Error("Not implemented");
   }
+
+  private async socketSend(data: string): Promise<void> {
+    if (!this.socket) {
+      throw new Error("No socket available");
+    }
+    await this.socket.connected;
+    this.socket.send(data);
```

Comment:
`this.socket.send` returns a promise that we ignore here. When this promise is rejected, we get crashes impossible to debug. Either do

```
await this.socket.send(data);
```

or 

```
return this.socket.send(data);
```

Both ways do more or less the same thing in this case.

Reply:
fixed!